client-dotnet/src/Controller/Controllers/EmController.cs

```diff
@@ -0,0 +1,280 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Client.Util;
+using Controller.Api;
+using Controller.Problem;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Controller.Controllers {
+
+  public class EmController : ControllerBase {
+
+    private readonly SutController _sutController;
+    private static string _baseUrlOfSut;
+
+    /*
+     Keep track of all host:port clients connect so far.
+     This is the mainly done for debugging, to check that we are using
+     a single TCP connection, instead of creating new ones at each request.
+     
+     However, we want to check it only during testing
+     */
+    //TODO: check
+    private static readonly ICollection<string> connectedClientsSoFar = new SynchronizedCollection<string> ();
+
+    private static readonly SemaphoreLocker _locker = new SemaphoreLocker ();
+
+    //ATTENTION: The relative path is from the SUT
+    private static readonly string htmlWarning = System.IO.File.ReadAllText ("../../../../client-dotnet/src/Controller/Resources/warning.html");
```

Comment:
I do not think this is correct. you are referring to a file on the hard-drive,  but that would not exist once the library is published as a DLL, isn't it? so, it should be read from the DLL, once packaged

Reply:
Yes you are right and it was my mistake. I modified this and now the html file gets copied into the SUT's bin directory after build. I think it is fixed now